components/eventing-controller/pkg/backend/jetstreamv2/utils.go

```diff
@@ -0,0 +1,222 @@
+	// TODO: Put this in the validation webhook
+	if sub.Spec.Types == nil || sub.Spec.Source == "" {
+		return []eventingv1alpha2.EventType{}, errors.New("event source and types must be provided")
+	}
+
+	uniqueTypes := getUniqueEventTypes(sub.Spec.Types)
+	var cleanEventTypes []eventingv1alpha2.EventType
+	for _, eventType := range uniqueTypes {
+		cleanType := eventType
+		var err error
+		if sub.Spec.TypeMatching != eventingv1alpha2.EXACT {
+			cleanType, err = getCleanEventType(eventType, cleaner)
+			if err != nil {
+				return []eventingv1alpha2.EventType{}, err
+			}
+		}
+		newEventType := eventingv1alpha2.EventType{
+			OriginalType: eventType,
+			CleanType:    cleanType,
+		}
+		cleanEventTypes = append(cleanEventTypes, newEventType)
+	}
+	return cleanEventTypes, nil
+}
+
+// isJsSubAssociatedWithKymaSub returns true if the given SubscriptionSubjectIdentifier and Kyma subscription
+// have the same namespaced name, otherwise returns false.
+func isJsSubAssociatedWithKymaSub(jsSubKey SubscriptionSubjectIdentifier, subscription *eventingv1alpha2.Subscription) bool {
+	return createKeyPrefix(subscription) == jsSubKey.NamespacedName()
+}
+
+//----------------------------------------
+// SubscriptionSubjectIdentifier utils
+//----------------------------------------
+
+// NamespacedName returns the Kubernetes namespaced name.
+func (s SubscriptionSubjectIdentifier) NamespacedName() string {
+	return s.namespacedSubjectName[:strings.LastIndex(s.namespacedSubjectName, separator)]
+}
+
+// ConsumerName returns the JetStream consumer name.
+func (s SubscriptionSubjectIdentifier) ConsumerName() string {
+	return s.consumerName
+}
+
+// NewSubscriptionSubjectIdentifier returns a new SubscriptionSubjectIdentifier instance.
+func NewSubscriptionSubjectIdentifier(subscription *eventingv1alpha2.Subscription, subject string) SubscriptionSubjectIdentifier {
+	cn := computeConsumerName(subscription, subject)          // compute the consumer name once
+	nn := computeNamespacedSubjectName(subscription, subject) // compute the namespaced name with the subject once
+	return SubscriptionSubjectIdentifier{consumerName: cn, namespacedSubjectName: nn}
+}
+
+// computeConsumerName returns JetStream consumer name of the given subscription and subject.
+// It uses the crypto/md5 lib to return a string of 32 characters as recommended by the JetStream
+// documentation https://docs.nats.io/running-a-nats-service/nats_admin/jetstream_admin/naming.
+func computeConsumerName(subscription *eventingv1alpha2.Subscription, subject string) string {
+	cn := subscription.Namespace + separator + subscription.Name + separator + subject
+	h := md5.Sum([]byte(cn)) // #nosec
+	return hex.EncodeToString(h[:])
+}
+
+// computeNamespacedSubjectName returns Kubernetes namespaced name of the given subscription along with the subject.
+func computeNamespacedSubjectName(subscription *eventingv1alpha2.Subscription, subject string) string {
```

Comment:
Do we have unit tests for each individual util methods: NewSubscriptionSubjectIdentifier, computeConsumerName, NamespacedName, createKeyPrefix, getStreamConfig etc.?

Reply:
Most of them are in place. Will add if anything is missing!